packages/apps/sendgrid/src/functios-to-sendgrid/abandoned-carts.ts

```diff
@@ -0,0 +1,74 @@
+import logger from 'firebase-functions/lib/logger';
+import api from '@cloudcommerce/api';
+// eslint-disable-next-line import/no-unresolved
+import { getFirestore } from 'firebase-admin/firestore';
+import { sgSendMail } from './utils';
+import parseCartToSend from './parse-data-to-send';
+
+export default async () => {
+  try {
+    logger.log('# Check abandoned carts');
+    const [application] = (await api.get(
+      'applications?app_id=129856&fields=hidden_data,data',
+    )).data.result;
+
+    const appData = {
+      ...application.data,
+      ...application.hidden_data,
+    };
+
+    let filterCartsDate = '';
+    const limitDate = new Date();
+    limitDate.setHours(limitDate.getHours() - (appData.abandoned_cart_delay || 3));
+    const lastNotificationSent = (await getFirestore().doc('sendGrid/lastCartSent').get()).data();
+    if (lastNotificationSent && lastNotificationSent.time) {
+      const lastNotifiedCart = new Date(lastNotificationSent.time);
+      filterCartsDate += `&create_at>=${lastNotifiedCart.toISOString()}`;
+    }
+
+    filterCartsDate += `&created_at<=${limitDate.toISOString()}`;
+
+    const [
+      { data: { result: abandonedCarts } },
+      { data: store },
+    ] = await Promise.all([
+      api.get(`carts?completed=false&available=true${filterCartsDate}`),
```

Comment:
```suggestion
      api.get(`carts?completed=false&available=true${filterCartsDate}&limit=100&sort=created_at`),
```

E se pá 100 é muito ainda, chegou a ver os limites de uso da API do SendGrid?

Reply:
Cheguei sim, na [doc](https://docs.sendgrid.com/api-reference/how-to-use-the-sendgrid-v3-api/rate-limits) eles falam que tem um limite para cada recurso, que deveria ser verificado com metodo `get`, no recurso que será utilizado, que os `limits` estariam no `header` da resposta, no entando fiz uma request como na documentação para o recurso `send`, mas os headres da resposta não tiveram os parametros que foram citados.